[Weave] annotation queue docs-1869 (#2119)
wandb/docs: weave/guides/tracking/annotation-queues.mdx

---
title: "Set up annotation queues"
description: "Create annotation queues, route traces to domain experts, and export structured feedback."
---

## Overview

Annotation queues let you route selected traces to domain experts for structured review without requiring them to navigate the full Weave UI. As an engineer, you define what feedback is collected, select which traces require review, and later can export completed annotations for analysis or dataset creation.

- **Manual trace scoring**: Have SMEs rate model outputs on correctness, quality, or style.
- **Failure analysis**: Annotate failure modes (hallucinations, refusals, loops) to understand where your model breaks.
- **Domain expert review**: Enable medical, legal, or safety experts to review content with a task-focused interface.
- **Dataset creation**: Turn annotated traces into evaluation or training datasets.

## End-to-end workflow

> As the engineer, you are responsible for the following lifecycle:

Reviewer: nit: The term "engineer" could be ambiguous—does it encompass roles like Project Manager, Data Analyst, or QA? Often, "engineer" is interpreted narrowly to mean software development engineers specifically. Should we consider a more inclusive term to cover these roles broadly?

Alternatively, is this feature intentionally scoped to "engineer" in its strictest sense? If so, I suggest peer-reviewing this definition and make sure folks are on the same page.

Author: this is a great point - i will have a chat with my team about 'personas' and how we refer to our audiences.

Author: Advice was to avoid naming them unless necessary. I reworked the two sentences on this page referring to 'engineer' so that they no longer needed to be labeled.

1. Define annotation fields and create an annotation queue.
2. Load traces into the queue for review.
3. Monitor progress while domain experts complete reviews.
4. Filter and export completed annotations.

## Define annotation fields

Annotation fields define the feedback that the annotator provides for each trace item. Fields are reusable across queues and projects.

Field types include:
- Boolean judgments such as correctness or acceptability.
- Numeric or integer values such as quality or confidence.
- Categorical labels such as failure mode or intent.
- Free-form text for qualitative feedback.

> Define all fields before creating a queue so they can be selected during queue setup.

Reviewer: Nit: how about we highlight this with `<Tip>` ?

Author: I don't think it qualifies as a "Tip" since it's required - but if you'd prefer this statement be highlighted I can make it a Note.

Reviewer: Yes, lets high-lighten it with Note.

To create an annotation field:
1. Navigate to [wandb.ai](https://wandb.ai).

> 2. In the Weave sidebar menu, click **Annotate**.

Reviewer: Friendly reminder: The **Annotate** might be hidden in the `More` menu item in case a project contains both models and weave traces usage.

Author: good point, will add this.

Author: I only added it on the first one though, just like only the first one indicates to 'navigate to wandb.ai', the rest of the procedures on the same page assume that.

3. In the tab bar, click the **Fields** tab.
4. In the Fields table toolbar, click **New Field**.
5. In the **Create annotation field** modal dialog, configure:
    - **Type**: Boolean, Integer, Number, String, or categorical options
    - **Name** 
    - **Description** (Optional)
6. Click **Create annotation field** to save the field.

<Frame>
  ![Annotation field creation dialog with inputs for name, description, and field type, used to define the schema annotators complete when reviewing trace items.](./imgs/AnnotationCreateField.png)
</Frame>

## Create an annotation queue

An annotation queue consists of:
- A set of annotation fields.
- Guidelines that provide task instructions for annotators.
- A collection of trace items awaiting review.

To create an annotation queue:
1. In the Weave sidebar menu, click **Annotate**.
2. In the tab bar, click the **Queues** tab.
3. In the Queues table toolbar, click **Create Queue**.
4. In the **Create Annotation Queue** modal dialog, configure:
    - **Queue name**: This is the queue name the annotator will select to complete their work.
    - **Guidelines** (Optional): Any additional instructions for the annotator.
5. Click **Next**.

> 6. Click **Manage fields** and choose what Annotation Fields should be included in this review work. All existing Annotation Fields for the Project are available for selection.

Reviewer: Nit: I guess we probably need some screenshot for this section. The other section has, this one doesn't which might render the doc unbalanced.

Author: Generally, we (Docs) try to avoid (or are trying more to avoid) screenshots for the sake of screenshots.  They are hard to maintain and know when they are out of date.  Thus, I had some extra screenshots but ultimately chose to hold back the ones that I didn't think added true value (e.g. showed something non-intuitive).
That being said, if you or others really feel that additional ones are needed, we can certainly add them.

Reviewer: Sure, as long as you are intentionally avoid doing that for a reason. It is good.

7. After you have selected all the Fields for the queue, click **Create Queue** to save the queue.

All annotation queues for the project are listed in the Annotation Queues page.
<Frame>
  ![Annotation Queues page displaying a table of queues in the project, including queue names, descriptions, and review progress to track annotation workflows.](./imgs/AnnotationQueues.png)
</Frame>

Creating an annotation queue defines the fields and guidelines for evaluation, but you still need to add traces to the queue to identify what data should be evaluated.

### Add traces to a queue

Traces can be added to an annotation queue directly from the **Traces** page:
1. In the Weave sidebar menu, click **Traces**.
2. In the Traces table toolbar, filter traces as needed (such as by hallucination scores, failure modes, or specific ops).
3. In the table, select the traces you want annotated.
4. In the table's action bar, click **Add to queue** to add the selected rows to an annotation queue.
<Frame>
  ![Traces table with multiple rows selected and a bulk action bar showing actions that apply to the selected rows, such as Add to queue and Add to dataset.](./imgs/AnnotationAddTraces.png)
</Frame>

> 5. In the **Add to annotation queue** modal dialog, configure:

Reviewer: Nit: maybe we need a screenshot of this modal dialog.

Author: See above about screenshots - trying to not add lots unless they help explain something that might not be obvious.  I thought the modal dialog was intuitive and easy to follow. if there's some aspect you think a screenshot may make more clear, let's discuss.

Reviewer: Sounds good.

    - **Select trace data to display**: Select the Weave trace data elements to display to the annotator as they are evaluating the results.
        - **Inputs**: Select which trace input fields to show during annotation.
        - **Outputs**: Select which trace output fields to show during annoation.
6. Click **Add ***[Count]*** traces to annotation queue** to assign these traces as a part of the annotation queue review.

When adding traces, you control which trace inputs and which outputs or model responses are reviewed. This way you can present annotators with only the context needed to make accurate judgments.

## Monitor review progress

Once the annotation queue has been created, and traces added, share the queue name with your annotator for them to begin their review. See [Review items in an annotation queue](/weave/guides/tracking/annotation-review) for details on the review process.

To share a direct link to the annotation queue with an annotator:
1. In the Weave sidebar menu, click **Annotate**.
2. In the tab bar, click the **Queues** tab.
3. In the Annotation Queues table, click the name of your queue to open the queue items.

> 4. In the Queue header bar, click the link button to copy a direct link to this queue.  You can also copy the URL from the browser address bar.

Reviewer: It might be hard to locate the link button simply by reading. can we have a screenshot to better indicate its location?

Author: Yep, will do.

---
title: "Set up annotation queues"
description: "Create annotation queues, route traces to domain experts, and export structured feedback."
---

## Overview

> Annotation queues let you route selected traces to domain experts for structured review without requiring them to navigate the full Weave UI. You define what feedback is collected, select which traces require review, and later can export completed annotations for analysis or dataset creation.

Reviewer 2: ```suggestion
Annotation queues let you route selected traces to domain experts for structured review without requiring them to navigate the full Weave UI. You define what feedback is collected, select which traces require review, and later can export completed annotations for analysis or dataset creation. Use cases can include:
```
Doesn't have to be this exact clause, but I think we should introduce what the bullet points are.

- **Manual trace scoring**: Have SMEs rate model outputs on correctness, quality, or style.
- **Failure analysis**: Annotate failure modes (hallucinations, refusals, loops) to understand where your model breaks.
- **Domain expert review**: Enable medical, legal, or safety experts to review content with a task-focused interface.
- **Dataset creation**: Turn annotated traces into evaluation or training datasets.

## End-to-end workflow

The following workflow summarizes how to use annotation queues to obtain reviews:

1. Define annotation fields.
2. Create an annotation queue.
3. Load traces into the queue for review.
4. Monitor progress while domain experts complete reviews.
5. Filter and export completed annotations.

> ## Define annotation fields

Reviewer 2: Feel like we need to denote somehow that we're starting the workflow here, because the prior two sections are informational and then we just kinda jump into the first sub-task of the setup without any warning.

I would approach it by adding a new header above "Define annotation fields" with an introductory sentence, then set the remaining sub-tasks as H3s below it, but I also see you're using H3s to introduce other nested-tasks, which would nest them further as H4s, and I don't know how the TOC on the right would handle that. 

Maybe just add a sentence at the beginning of this section that says, "To start creating an annotation queue, you need to define some annotation fields first. Annotation fields define the..."

Annotation fields define the feedback that the annotator provides for each trace item. Fields are reusable across queues and projects.

Field types include:
- Boolean judgments such as correctness or acceptability.
- Numeric or integer values such as quality or confidence.
- Categorical labels such as failure mode or intent.
- Free-form text for qualitative feedback.

> <Note>

Reviewer 2: I don't think this needs to be a note. It's necessary to the workflow and not really a caveat, pro tip, or additional insightful information. 

Also, sorry in advance, I work to eliminate Note boxes as if they've offended me because there's a growing amount of evidence that users just gloss over them. It's totally a byproduct of everyone's brains being trained to navigate ad-ridden minefields of news story content. 😅

Author: concur, lol

Define all fields before creating a queue so they can be selected during queue setup.
</Note>

To create an annotation field:
1. Navigate to [wandb.ai](https://wandb.ai) and select your project.
2. In the Weave project sidebar, click **Annotate**.  If you don't see **Annotate**, it might be nested in the menu under **More**.
3. In the tab bar, click the **Fields** tab.
4. In the Fields table toolbar, click **New Field**.
5. In the **Create annotation field** modal dialog, configure:

>     - **Type**: Boolean, Integer, Number, String, or categorical options

Reviewer 2: Should we describe these fields? I know they're kinda self-explanatory. I always do, but I've often wondered whether we actually should.

The Google style guide has a slightly different approach in the last example here: https://developers.google.com/style/ui-elements#term-windows

Author: not sure what on that link I was supposed to focus on.  regarding explaining boolean, integer, etc,... I'd lean towards that being 'expected knowledge' for our audience.  Though yeah, if we decide to align on including that level of detail I don't feel strongly either way.  (tho i think i'll leave this this way for now and circle back if we decide we should)

Author: or maybe you meant the name and description.. yeah, seemed too obvious, but anyway, i went and added them now for completeness

    - **Name** 
    - **Description** (Optional)
6. Click **Create annotation field** to save the field.

<Frame>
  ![Annotation field creation dialog with inputs for name, description, and field type, used to define the schema annotators complete when reviewing trace items.](./imgs/AnnotationCreateField.png)
</Frame>

## Create an annotation queue

An annotation queue consists of:
- A set of annotation fields.
- Guidelines that provide task instructions for annotators.
- A collection of trace items awaiting review.

To create an annotation queue:
1. In the Weave project sidebar, click **Annotate**.
2. In the tab bar, click the **Queues** tab.
3. In the Queues table toolbar, click **Create Queue**.
4. In the **Create Annotation Queue** modal dialog, configure:

>     - **Queue name**: This is the queue name the annotator will select to complete their work.

Reviewer 2: ```suggestion
    - **Queue name**: This is the queue name the annotator selects to complete their work.
```

    - **Guidelines** (Optional): Any additional instructions for the annotator.
5. Click **Next**.

> 6. Click **Manage fields** and choose what Annotation Fields should be included in this review work. All existing Annotation Fields for the Project are available for selection.

Reviewer 2: ```suggestion
6. Click **Manage fields** and choose what Annotation Fields to include in this review work. All existing Annotation Fields for the Project are available for selection.
```

7. After you have selected all the Fields for the queue, click **Create Queue** to save the queue.

All annotation queues for the project are listed in the Annotation Queues page.
<Frame>
  ![Annotation Queues page displaying a table of queues in the project, including queue names, descriptions, and review progress to track annotation workflows.](./imgs/AnnotationQueues.png)
</Frame>

Creating an annotation queue defines the fields and guidelines for evaluation, but you still need to add traces to the queue to identify what data should be evaluated.

### Add traces to a queue

> Traces can be added to an annotation queue directly from the **Traces** page:

Reviewer 2: ```suggestion
To add traces to an annotation queue directly from the **Traces** page:
```
Makes this a little more active

1. In the Weave project sidebar, click **Traces**.
2. In the Traces table toolbar, filter traces as needed (such as by hallucination scores, failure modes, or specific ops).
3. In the table, select the traces you want annotated.
4. In the table's action bar, click **Add to queue** to add the selected rows to an annotation queue.
<Frame>
  ![Traces table with multiple rows selected and a bulk action bar showing actions that apply to the selected rows, such as Add to queue and Add to dataset.](./imgs/AnnotationAddTraces.png)
</Frame>
5. In the **Add to annotation queue** modal dialog, configure:

>     - **Queue Name**: In the dropdown, select the name of the existing queue to add these traces to.

Reviewer 2: ```suggestion
    - **Queue Name**: In the list, select the name of the existing queue to add these traces to.
```
Google Style Guide recommends not using dropdown and using either "list" or "menu' instead: https://developers.google.com/style/word-list#drop-down

Author: i made the change, cuz google says so, but let the record show I strongly dislike this :)
saying 'list' makes me think it is something displayed on the page I should look for, 'dropdown' , to me, implies I need to take an action in order to see it (e.g. click)....

    - **Select trace data to display**: Select the Weave trace data elements to display to the annotator as they are evaluating the results.
        - **Inputs**: Select which trace input fields to show during annotation.
        - **Outputs**: Select which trace output fields to show during annoation.
6. Click **Add ***[Count]*** traces to annotation queue** to assign these traces as a part of the annotation queue review.

When adding traces, you control which trace inputs and which outputs or model responses are reviewed. This way you can present annotators with only the context needed to make accurate judgments.

## Monitor review progress

> Once the annotation queue has been created, and traces added, share the queue name with your annotator for them to begin their review. See [Review items in an annotation queue](/weave/guides/tracking/annotation-review) for details on the review process.

Reviewer 2: ```suggestion
Once you have created the annotation queue and added traces to it, share the queue name with your annotator for them to begin their review. See [Review items in an annotation queue](/weave/guides/tracking/annotation-review) for details on the review process.
```

To share a direct link to the annotation queue with an annotator:
1. In the Weave project sidebar, click **Annotate**.
2. In the tab bar, click the **Queues** tab.
3. In the Annotation Queues table, click the name of your queue to open the queue items.
4. In the Queue header bar, click the link button to copy a direct link to this queue.  You can also copy the URL from the browser address bar.

  ![Copy link to the annotation queue in the Annotation Queue header bar](./imgs/AnnotationLinkToQueue.png)

In the Annotation Queues table, the **State** column indicates reviewing progress:
- **Not started**: Queue has items but no annotations have been submitted.
- **In progress**: At least one item has been reviewed.
- **Completed**: All items have been reviewed.

In the Annotation Queues table, the **Calls with responses** column indicates the percent of items (out of the total number of **Calls**) that have at least one submitted review.

## Filter and export annotations

> Completed annotations are stored as structured metadata on traces.

Reviewer 2: ```suggestion
Weave stores completed annotations as structured metadata on traces.
```

You can:
- Filter traces by queue assignment and annotation completion.
- Save filtered views for reuse.
- Export annotated traces to datasets for evaluation or training workflows.

This connects expert human feedback directly to model evaluation and iteration.

### Filter annotated traces

> 1. In the Weave project sidebar, click **Traces**.

Reviewer 2: Need an introductory clause here to set off these directions. I know it kinda seems redundant, but it's the style used across the entire website

2. In the Traces table toolbar, click **Filter**.
3. Add three values to a filter row:
    - For **Column**, type "Queue", then press **Enter**.
    - For the second dropdown, choose **Text: "is"**.
    - For **Select a queue**, choose your annotation queue name.
4. To also filter on 'completed' queue items only, click **+ Add Filter**:

>     - For **Column**, type "feedback".  A popup will populate with Annotations and include your Annotation Field names.  Choose a required Field from your queue.

Reviewer 2: ```suggestion
    - For **Column**, type "feedback".  A dialog populates with Annotations and includes your Annotation Field names.  Choose a required Field from your queue.
```
https://developers.google.com/style/word-list#pop-up

</Frame>
4. Filter rows are automatically applied; click elsewhere in the page for the filter entry to close.
5. (Optional) Save as a view for quick access. In the Traces table header, click **Save View**.

### Export annotated traces to datasets

You can export annotated traces either through the UI or programmatically, depending on how you plan to use the data.

#### Add annotated traces to a dataset

Select annotated traces and click **Add to Dataset** to include expert labels in your eval or training data.

1. In the Weave project sidebar, click **Traces**.
2. In the Traces table, select the traces that you want to export.
3. In the table toolbar, click **Add to dataset**. Follow the on-screen prompts to complete the addition.

To learn more about using datasets, see [Collect and track datasets](/weave/guides/core-types/datasets).

#### Access annotations programmatically

If you want to integrate annotations programmatically, you must know your project name and queue ID:
 - **Project**: The W&B project name (can be project or team/project). If you don’t specify a W&B team (such as "team/project"), your default team is used.

>  - **Queue ID**: The queue ID is the unique identifier for an annotation queue.

Reviewer 2: ```suggestion
 - **Queue ID**: The annotation queue's unique identifier.
```

To find the queue ID:
1. In the **Annotation Queues** table, select the name of the queue to open its items.
2. Copy the ID from the end of the page URL.

**Example**
```
https://wandb.ai/.../annotation-queues/019c0f63-7acb-7497-8f87-08873368fcd4
```

In this example, the queue ID is:

019c0f63-7acb-7497-8f87-08873368fcd4

You can iterate through the traces (calls) in your queue using the following code.

> ```python

Reviewer 2: ```suggestion
```python lines
```

Add line numbers to the code example.
